src/System.Private.CoreLib/shared/System/BadImageFormatException.cs

```diff
@@ -95,11 +95,10 @@ public override string ToString()
         {
             string s = GetType().ToString() + ": " + Message;
 
-            if (!string.IsNullOrEmpty(_fileName))
+            if (!string.IsNullOrEmpty(_fileName) && !Message.Contains(_fileName))
```

Comment:
Is this all that meaningful? While we discourage parsing, the current format is much more parsable for the filename.

Reply:
Assuming the filename is set and not just in the message. Ok, guess I need feedback on this one.

Apart from reducing noise, another possible advantage of reducing line count is that the parts of the output are closer together in a log produced from concurrent Interleaving outputs eg a build log or test run.